### PR TITLE
Add name field reference to PEP 508

### PR DESCRIPTION
**Pull request by gunungpw**
# Pull Request Check List

Resolves: #5267 

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

Add documentation for name field reference to [PEP 508](https://peps.python.org/pep-0508/#names)

### CODE REVIEW

File: docs/pyproject.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ The `tool.poetry` section of the `pyproject.toml` file is composed of multiple s
 ## name
 
 The name of the package. **Required**
+This should be valid name according to specifications in [PEP 508 - Names](https://peps.python.org/pep-0508/#names)
```
Comment (Secrus) [inline docs/pyproject.md:19]:
```suggestion

This should be a valid name according to specifications in [PEP 508 - Names](https://peps.python.org/pep-0508/#names)
```

**Review by Secrus — CHANGES_REQUESTED** [review]
Small formatting and grammar changes.

File: docs/pyproject.md (hunk 1 above)
Comment (gunungpw) [reply docs/pyproject.md]: My bad english not my first language 👍

**Review by Secrus — DISMISSED** [review]

File: docs/pyproject.md [hunk 2]
```diff
@@ -17,6 +17,9 @@ The `tool.poetry` section of the `pyproject.toml` file is composed of multiple s
 
 The name of the package. **Required**
 
+This should be a valid name according to specifications in [PEP 508 - Names](https://peps.python.org/pep-0508/#names)
```
Comment (neersighted) [inline docs/pyproject.md:20]:
```suggestion
This should be a valid name as defined by [PEP 508](https://peps.python.org/pep-0508/#names).
```

**Review by neersighted — CHANGES_REQUESTED** [review]

**Review by abn — APPROVED** [review]

**Pull request merged**